src/messaging/ExchangeContext.cpp

```diff
@@ -531,34 +535,49 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
         MessageHandled();
     });
 
-    if (mDispatch.IsReliableTransmissionAllowed() && !IsGroupExchangeContext())
+    if (GetSessionHandle()->AllowsMRP())
     {
-        if (!msgFlags.Has(MessageFlagValues::kDuplicateMessage) && payloadHeader.IsAckMsg() &&
-            payloadHeader.GetAckMessageCounter().HasValue())
+        if (mDispatch.IsReliableTransmissionAllowed() && !IsGroupExchangeContext())
```

Comment:
If AllowsMRP(), then IsGroupExchangeContext() is false.

Reply:
The whole check of `if (mDispatch.IsReliableTransmissionAllowed() && !IsGroupExchangeContext())` can be removed it seems.

Reply:
No, `mDispatch.IsReliableTransmissionAllowed()` needs to stay.  That's actually there for some test stuff that overrides default behavior.

Reply:
It is there. I have removed the `!IsGroupExchangeContext()` part.